test/1_stdlib/ErrorHandling.swift

```diff
@@ -340,4 +340,42 @@ ErrorHandlingTests.test("ErrorHandling/Collection map") {
   }
 }
 
+/* when b == 2 we are losing elements. insertion sort throws error
+in between while sorting so loop is not able to complete and swap the elements and array mutates
+to [2,3,3]
+ErrorHandlingTests.test("ErrorHandling/sort") {
+  var collection = [3, 2, 1]
+  do {
+    try collection.sort { (a, b) throws -> Bool in
+      print("collection before error \(collection)")
+      print("a \(a) \(b)")
+
+      if b == 2 {
```

Comment:
Yes, and in 99% of cases a partly sorted collection is useless to you and you have no alternative way to do the same sort, so in 99% of cases you’ll be sorting something you’re willing to lose, like a copy of the original collection.  

By the way, if you *did* have a fallback for a failing comparison, you’d just make your comparison fall back to it rather than aborting the sort and starting over again.